evm/src/cpu/kernel/asm/memory.asm

```diff
@@ -0,0 +1,27 @@
+// Load a byte from the given segment of the current context's memory space.
+// Note that main memory values are one byte each, but in general memory values
+// can be 256 bits. This macro deals with a single address (unlike MSTORE), so
+// if it is used with main memory, it will load a single byte.
+%macro mload_current(segment)
```

Comment:
I see how this could be confusing, since it sounds like `MLOAD` but doesn't have the same behavior of combining 32 values. But I think `MLOAD8` could also be confusing, since certain segments (like the stack) contain values larger than 8 bits, unlike main memory. E.g. `%mload_current(@SEGMENT_STACK)` could give 256 bits. (Not sure if we'd ever have a reason to use this macro with the stack, but we could also use it with general Kernel memory which could contain large values.)

Maybe we could call it `%mload_single_current` to emphasize that it deals with a single address, but it might be nice to keep it short since it may be used a lot.